lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -79,12 +79,14 @@ function extractVersion({
   versionSubContent: string;
 }): VersionExtract {
   if (isVersionPointer(version)) {
+    // This is to support dot annotation version reference
+    const parsedVersion = version.ref.replaceAll('.', '-').replaceAll('_', '-');
```

Comment:
```suggestion
    const parsedVersion = version.ref.replace(/[._]/g, '-');
```

https://jsben.ch/KY0Cv